datafusion/core/tests/sqllogictests/src/engines/postgres/mod.rs

```diff
@@ -0,0 +1,197 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::sync::Arc;
+use std::time::Duration;
+
+use async_trait::async_trait;
+use log::info;
+use sqllogictest::{ColumnType, DBOutput};
+use testcontainers::core::WaitFor;
+use testcontainers::images::generic::GenericImage;
+use tokio::task::JoinHandle;
+
+pub struct Postgres {
```

Comment:
This implementation is a full copy of https://github.com/risinglightdb/sqllogictest-rs/blob/main/sqllogictest-bin/src/engines/postgres.rs from the `sqllogictest-bin`. It could make sense to make the `sqllogictest-bin` postgres client a public module https://github.com/risinglightdb/sqllogictest-rs/blob/main/sqllogictest-bin/src/engines/mod.rs#L1 and use the cargo dependency.

Reply:
I think coping it from sqllogictest-rs would be ok for now; However,  @xxchan  has been very interested in helping to make sqllogictest easier to use / refactor -- if they were interested we could propose an upstream PR to put this logic somewhere (perhaps in the main sqllogictest-rs crate itself or a sqllogictest-postgres crate 🤔 )

Reply:
The implementation changed to provide the same type conversions for both Datafusion and Postgres.